src/history/hs_rec.c

```diff
@@ -296,7 +288,17 @@ __wt_hs_insert_updates(
     WT_RET(__wt_curhs_open(session, NULL, &hs_cursor));
     F_SET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
 
-    __wt_modify_vector_init(session, &modifies);
+    __wt_update_vector_init(session, &updates);
+    /*
+     * We use another stack to store the out-of-order timestamp updates (including updates without a
+     * timestamp). We walk the update chain from the newest to the oldest. Once an out-of-order
+     * timestamp update is detected, if it has lower timestamp than the head of the stack, it is
```

Comment:
is detected, if it has lower timestamp than the head ->is detected, and it has a lower timestamp than the head